kahuna/public/js/services/api/collections-api.js

```diff
@@ -64,6 +66,41 @@ apiServices.factory('collections', ['mediaApi', function (mediaApi) {
         return Promise.all(promises);
     }
 
+    function collectionsEquals(collectionsA, collectionsB) {
+        return angular.equals(
+            collectionsA.sort(),
+            collectionsB.sort()
+        );
+    }
+
+    function getCollectionsIds(imageCollections) {
+        const collectionsIds = [];
+        imageCollections.data.forEach(e => collectionsIds.push(e.pathId));
+        return collectionsIds;
+    }
+
+    function untilCollectionsEqual(image, expectedCollections) {
+        return image.get().then(apiImage => {
```

Comment:
I'm guilty of having suggested that to @tsop14. It does affect the image state doesn't it, insofar as it requires updating the list of collections found on the image?

I'm going to look into the long-fabulated way to do this more properly without read-after-write and annoying polling, but I think this is the simplest way for now. If you have any other thoughts or suggestions, please share!